src/types/mod.rs

```diff
@@ -50,24 +60,43 @@ pub enum BlockStatus {
 #[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct BlockDetails {
+    /// Number of the block.
     pub number: u64,
+    /// Corresponding L1 batch number.
     pub l1_batch_number: u64,
+    /// Address of the operator who committed the block.
     pub operator_address: Address,
+    /// Version of the ZKsync protocol the block was committed under.
     pub protocol_version: Option<String>,
+    /// Unix timestamp when the block was committed.
     pub timestamp: u64,
+    /// Number of L1 transactions included in the block.
     pub l1_tx_count: u64,
+    /// Number of L2 transactions included in the block.
     pub l2_tx_count: u64,
+    /// Root hash of the block's state after execution.
```

Comment:
This probably needs to be rephrased. Block doesn't have state. Is it a hash of L2 block? IIRC L2 blocks have rolling hashes over the transactions in them, while L1 root hash for batches is a hash of the blockchain state.

Reply:
Great catch, you're right it's a L2 block hash. Fixed the annotation.